abcclassroom/github.py

```diff
@@ -61,6 +59,9 @@ def get_access_token():
     return access_token
 
 
+# TODO: document this function with parameters and returns but also
+# should it provide a user friendly message based upon where it fails or
+# doesnt fail?
 def check_git_ssh():
     """Tests that ssh access to GitHub is set up correctly on the users
     computer.
```

Comment:
hey there @kcranston me again. i think i'm getting a better handle on how this works now but i have a question about this function. The comment below says " We ALWAYS get here". This is true - it seems like [when you use check=True it ](https://docs.python.org/3/library/subprocess.html) it returns an error w a non zero exit code.

My question is - why do we use check=True here? Without it, it just seems to run and return:
```
Out[6]: CompletedProcess(args=['ssh', '-T', 'git@github.com'], returncode=1, stdout='', stderr="Hi lwasser! You've successfully authenticated, but GitHub does not provide shell access.\n")
```

is this potentially because i may see different behavior when running at the command line? just trying to understand as i'm writing tests. it's more complex to route through all of these try/except blocks and more complex to test this way but perhaps there is a reason that i just don't understand to force that error to be thrown when using subprocess w/ `check=True`

Reply:
This particular case is strange, because the subprocess call always returns a non-zero exit code. So, we need to check the specific error message from git to see what's going wrong. If we re-write with check=False, then I think the _call_git function won't return the git messages (because it won't catch a CalledProcessError).

Reply:
ahhh so it's because without raising some sort of error, we can't parse what happened. 
Oh also i just thought of this - because it's a subprocess call would it also just fail or pass quietly without `check=True` ? is that essentially what you're saying? i think that makes sense to me if it's running at the command line, how could it catch a failure.

i just tried it with a fake git command and i see it "passed" but  failed at the CLI

Reply:
ok  i understand @kcranston i'll work on testing through all of the  try/excep pieces!! this is super helpful!